src/registrar/assets/src/js/getgov-admin/domain-request-form.js

```diff
@@ -629,12 +629,60 @@ export function initRejectedEmail() {
     });
 }
 
+
+/**
+ * A function that handles the suborganzation and requested suborganization fields and buttons.
+ * - Fieldwise: Hooks to the sub_organization, suborganization_city, and suborganization_state_territory fields.
+ *   On change, this function checks if any of these fields are not empty: 
+ *   sub_organization, suborganization_city, and suborganization_state_territory.
+ *   If they aren't, then we show the "clear" button. If they are, then we hide it because we don't need it.
+ * 
+ * - Buttonwise: Hooks to the #clear-requested-suborganization button.
+ *   On click, this will clear the input value of sub_organization, suborganization_city, and suborganization_state_territory.
```

Comment:
The "clear all" button is convenient, but it has too much visual prominence and conflicts with the Save buttons on the page. Let's reduce the visual footprint by replacing that with a simple text button to the right of the "Requested suborganization" field that reads: Clear requested suborganization

This should be preceded by an X icon to match patterns we established with the Copy button.
![Screenshot 2024-12-27 at 2 58 45 PM](https://github.com/user-attachments/assets/9fbdc793-368d-403f-8537-c0ccff786b83)